astropy/coordinates/representation.py

```diff
@@ -1029,8 +1037,18 @@ def get_xyz(self, xyz_axis=0):
         Returns
         -------
         xyz : `~astropy.units.Quantity`
-            With dimension 3 along ``xyz_axis``.
+            With dimension 3 along ``xyz_axis``.  Note that, if possible,
+            this will be a view.
         """
+        if self._xyz is not None:
+            if self._xyz_axis == xyz_axis:
+                return self._xyz
+            else:
+                return np.moveaxis(self._xyz, self._xyz_axis, xyz_axis)
+
+        # Create combined array.  TO DO: keep it in _d_xyz for repeated use?
+        # But then in-place changes have to cancel it. Likely best to
+        # also update components.
         return _combine_xyz(self._x, self._y, self._z, xyz_axis=xyz_axis)
```

Comment:
Is `_combine_xyz` implemented as a function mainly so that it can also be called from the tests?  That's all I see it used for, but if there's some other place it's used it might be best re-done as a method and have the caching be done there instead?  It's a bit confusing now that the caching and combining are spread over two places.

Reply:
@eteq - `_combine_xyz` is called from both `CartesianRepresentation` and `CartesianDifferential`. I in fact considered a `CartesianMixin` class which would have this method and which would also cache the result. But I ended up being unsure how to deal with in-place changes (strictly not meant to be done, but we have an documentation example where we show how to do it... it would screw up transformations). So, I decided to stick to the simple change for this PR, where all that is done is to keep any input array that already had 3 components.